nilearn/plotting/img_plotting.py

```diff
@@ -595,6 +643,168 @@ def plot_roi(roi_img, bg_img=MNI152TEMPLATE, cut_coords=None,
+            Matplotlib figure used or its number. If None is given, a
+            new figure is created.
+        axes : matplotlib axes or 4 tuple of float: (xmin, ymin, width, height), optional
+            The axes, or the coordinates, in matplotlib figure space,
+            of the axes used to display the plot. If None, the complete
+            figure is used.
+        title : string, optional
+            The title displayed on the figure.
+        annotate: boolean, optional
+            If annotate is True, positions and left/right annotation
+            are added to the plot.
+        draw_cross: boolean, optional
+            If draw_cross is True, a cross is drawn on the plot to
+            indicate the cut plosition.
+        black_bg: boolean, optional
+            If True, the background of the image is set to be black. If
+            you wish to save figures with a black background, you
+            will need to pass "facecolor='k', edgecolor='k'" to pylab's
+            savefig.
+        cmap: matplotlib colormap, optional
+            The colormap for the atlas maps
+        vmin: float
+            Lower bound for plotting, passed to matplotlib.pyplot.imshow
+        vmax: float
+            Upper bound for plotting, passed to matplotlib.pyplot.imshow
+        alpha: float between 0 and 1
+            Alpha value sets the transparency of the color fillings inside the
+            contours.
+    """
+    display = plot_anat(anat_img, cut_coords=cut_coords,
+                        display_mode=display_mode,
+                        figure=figure, axes=axes, title=title,
+                        annotate=annotate, draw_cross=draw_cross,
+                        black_bg=black_bg, **kwargs)
+
+    maps_img = _utils.check_niimg_4d(maps_img)
+    n_maps = maps_img.shape[3]
+
+    allowed_view_types = list(('auto', 'contours', 'filled_contours', 'continuous'))
+    if not (isinstance(view_type, _basestring) or
+            view_type not in allowed_view_types):
+        message = ('view_type option should be given '
+                   'either of these {0} ').format(allowed_view_types)
+        raise ValueError(message)
+
+    cmap = plt.cm.get_cmap(cmap)
+    # Build a custom colormap for displaying contours
+    color_list = cmap(np.linspace(0, 1, n_maps))
+
+    if view_type == 'auto':
+        if n_maps > 20:
+            view_type = 'contours'
+        elif n_maps > 10:
+            view_type = 'filled_contours'
+        else:
+            view_type = 'continuous'
+
+    if threshold is None:
+        # it will use default percentage,
+        # for a nicer look avoiding maximum overlaps for visualization
+        if view_type == 'contours':
+            correction_factor = 1
+        elif view_type == 'filled_contours':
+            correction_factor = .8
+        else:
+            correction_factor = .5
+        threshold = "%f%%" % (100 * (1 - .2 * correction_factor / n_maps))
+
+    if isinstance(threshold, list):
```

Comment:
Also supporting iterables in general, without an extremely good reason, is a pain in the ass as we already experienced elsewhere in nilearn. For example, this code will break with an iterable because you consume the iterable at least twice, `len(threshold) and `enumerate(zip(..., threshold))`.